cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCache.kt

```diff
@@ -893,57 +1022,115 @@ class SqlCache(
+   * @return hash of the body.
+   */
+  private fun getBodyHash(body: String?): String? {
     if (body.isNullOrBlank()) {
       return null
     }
-    return try {
-      val digest = MessageDigest.getInstance("SHA-256")
-        .digest(body.toByteArray())
-      digest.fold("") { str, it ->
-        str + "%02x".format(it)
-      }
-    } catch (e: Exception) {
-      log.error("error calculating hash for body: $body", e)
-      null
-    }
+    return getSha256Hash(body)
+  }
+
+  /**
+   * Returns a hash of the id given.
+   *
+   * @param id The id to hash.
+   * @return hash of the id.
+   */
+  private fun getIdHash(id: String): String {
+    return getSha256Hash(id)
+  }
+
+  /**
+   * Returns a hash of the agent given.
+   *
+   * @param agent the agent to hash.
+   * @return hash of the agent.
+   */
+  private fun getAgentHash(agent: String): String {
+    return getSha256Hash(agent)
   }
 
-  private fun getHashIds(type: String, agent: String?): List<HashId> {
+  /**
+   * Gets a SHA-256 hash value for a given String.
+   *
+   * @param str the String to hash.
+   * @return the hashed value.
+   */
+  private fun getSha256Hash(str: String): String {
+    return Hashing.sha256().hashBytes(str.toByteArray()).toString()
+  }
+
+  /**
+   * Returns the body hashes and id hashes stored for the given type and agent hash.
+   *
+   * @param type the type of the records to retrieve.
+   * @param agentHash the agent hash of the records to retrieve.
+   * @return list of resource entries with body hash and id hash populated.
+   */
+  private fun getBodyHashesAndIdHashes(type: String, agentHash: String?): List<ResourceEntry> {
     return withRetry(RetryCategory.READ) {
       jooq
-        .select(field("body_hash"), field("id"))
+        .select(field("body_hash"), field("id_hash"))
         .from(table(sqlNames.resourceTableName(type)))
         .where(
-          field("agent").eq(agent)
+          field("agent_hash").eq(agentHash)
         )
         .fetch()
-        .into(HashId::class.java)
+        .into(ResourceEntry::class.java)
     }
   }
 
-  private fun getRelationshipKeys(type: String, sourceAgent: String): MutableList<RelId> {
+  /**
+   * Returns the following identifying fields of records matching the given rel agent hash from the
+   * relationship table of the given type: uuid, id hash, rel id hash, and rel agent hash.
+   *
+   * @param type the type of the relationship table.
+   * @param relAgentHash the rel agent hash of the records whose identifying fields are retrieved.
+   * @return list of relationship entries with uuid, id hash, rel id hash, and rel agent hash
+   * populated.
+   */
+  private fun getRelationshipKeys(type: String, relAgentHash: String): MutableList<RelEntry> {
     return withRetry(RetryCategory.READ) {
       jooq
-        .select(field("uuid"), field("id"), field("rel_id"), field("rel_agent"))
+        .select(field("uuid"), field("id_hash"), field("rel_id_hash"), field("rel_agent_hash"))
```

Comment:
See comment above for more details, but if we return the full `rel_agent` here instead of just the hash, we can avoid constructing the reverse-lookup table above.